### PR TITLE
feat: add customindex metadata

### PR DESCRIPTION
**Pull request by jdkgabri**
## What does this pull request contain?

This pull request contains the addition of the `customindex`  metadata to the ` v55.json` file.

---

- [x] Added (for new features).
- [ ] Changed (for changes in existing functionality).
- [ ] Deprecated (for soon-to-be removed features).
- [ ] Removed (for now removed features).
- [ ] Fixed (for any bug fixes).
- [ ] Security (for vulnerability fixes).

## Explain your changes

---

This Pull Requests aims to add support in order detect and include added custom indexes in the result `package.xml`. This metadata type is not officially documented yet, so not sure if it is a new feature or an undocumented behaviour leak. Read more about it [here](https://salesforce.stackexchange.com/questions/376766/create-custom-index-via-metadata-api-now-possible.

Maybe it is possible to ask this internally with the Salesforce Metadata API Product team before accepting this PR.

## Does this close any currently open issues? 
No.

---

- [ ] Jest test added to check the fix is applied.

## Any particular element that can be tested locally

The inclusion of the CustomIndex elements in the output file package.xml when they are detected as added/modified by the command usage.

---

## Any other comments

---

## Where has this been tested?

---

**Operating System:**  Windows 11

**yarn version:** 1.22.17

**node version:** v16.13.0

**git version:** 2.33.1.windows.1

**sfdx version:** sfdx-cli/7.160.0 win32-x64 node-v16.16.0

**sgd plugin version:** 5.3.0

### CODE REVIEW

**Comment by scolladon** [comment]
Interesting, thanks for this awesome addition to the plugin @jdkgabri 

I found in the documentation something about the [index](https://developer.salesforce.com/docs/atlas.en-us.238.0.api_meta.meta/api_meta/meta_index.htm). It is very close from what is added but it is not custom index and works only for Big Object.

This custom index works for every kind of objects right ? Custom objects and standard objects ?

**Comment by jdkgabri** [comment]
Yes, custom index works for Standard and Custom Objects, with both, standard and custom fields. For example:

image

File: src/metadata/v55.json
Comment (scolladon) [inline src/metadata/v55.json:3]: **remark:** I think it should be added in the list of Custom Object children line [156](https://github.com/scolladon/sfdx-git-delta/pull/324/files#diff-f6449921ca6fbe6bc2cd8d44b67a557af1fc3778f5c7ee415ddc516e105014ceR156) as well

**Review by scolladon — CHANGES_REQUESTED** [review]
Nice addition, one thing remaining to be complete I guess

**Comment by scolladon** [comment]
> 

Ok thanks, I'm contacting the product to have more information

Have you been able to test this change ? Also using `--generate-delta` parameter ?

File: src/metadata/v55.json
Comment (jdkgabri) [reply src/metadata/v55.json:3]: Actually, customindex folder is not nested into the parent object folder. The structure is like:

├── default
    ├── customindex
    │    └── Account.Fax.indx-meta.xml
    └── objects
        └── Account
            └── fields
                └──  Fax.field-meta.xml

Based on this, I would believe it should not be added, but maybe I am misunderstanding something.

**Comment by jdkgabri** [comment]
> Have you been able to test this change ? Also using `--generate-delta` parameter ?

Actually I was just checking again and I am unable to get it working as expected. I though I tested it properly. I am investigating what can I be missing.

File: src/metadata/v55.json
Comment (scolladon) [reply src/metadata/v55.json:3]: Ok clear, thanks for the update

**Comment by scolladon** [comment]
Ok top

I'll test that locally as well

**Review by scolladon — APPROVED** [review]
LGTM

I'll wait for my test and for the feedback from the product before merging
I want to know it is a real feature

**Comment by jdkgabri** [comment]
Sure, it sounds like the good thing to do. Thanks!

**Comment by scolladon** [comment]
This PR is ready to be shipped.
CustomIndex is a real metadata and will stay.
It should be documented soon

**Comment by jdkgabri** [comment]
Great to know! Thanks for confirming it!

**Review by mehdisfdc — APPROVED** [review]
LGTM

**Comment by scolladon** [comment]
Thank you very much @jdkgabri 

New version [5.4.0](https://github.com/scolladon/sfdx-git-delta/releases/tag/v5.4.0) is just released with this enhancement !
Thanks for helping improving the plugin

Happy incremental deployment !!

**Comment by scolladon** [comment]
Thank you very much @jdkgabri

New version [5.4.0](https://github.com/scolladon/sfdx-git-delta/releases/tag/v5.4.0) is just released with this enhancement !
Thanks for helping improving the plugin

Happy incremental deployment !!

**Pull request merged**
